Makefile

```diff
@@ -0,0 +1,48 @@
+
+.PHONY: build test clean docker
+
+MICROSERVICES=build/release/device-coap
+.PHONY: $(MICROSERVICES)
+
+DOCKERS=docker_device_coap_c
+.PHONY: $(DOCKERS)
+
+VERSION=$(shell cat ./VERSION)
```

Comment:
Ironically, @iain-anderson and I [discussed](https://github.com/edgexfoundry-holding/device-coap-c/pull/1#discussion_r506400068) VERSION on the initial pull request for this repo. I understand that the current implementation, based on device-grove-c, is the expected practice. Iain, please confirm.

Reply:
Yes, the implementation in device-bacnet-c requires a VERSION file to be maintained in the repository which isn't ideal

Reply:
Thanks for confirming. I have a follow-up though. For both device-grove-c and device-coap-c the user must manually remove the VERSION file to ensure it is updated after a new tag is created. Would it make more sense to remove VERSION on a `make clean` since `make build` depends on it? I guess this is a minor point since the main concern is for automated builds which will do a repository checkout anyway. I ran into this issue while researching for this comment.

Reply:
agree, `make clean` should remove VERSION

Reply:
Addressed VERSION file cleanup in 21760e4

Reply:
@iain-anderson , why can't we use the standard way of setting Version that we have in the other makefiles?
```
VERSION=$(shell cat ./VERSION || echo 0.0.0)
```
The CI Pipeline creates the file on the fly base on tag. This allows us not to have a committed VERSION file, create one locally on the fly, etc. As stay consistent with all the other Makefiles.

Reply:
@lenny-intel the build using cmake+make ingests the version string via file. We could look into using other means but this would be across all of the C builds - maybe open an issue for consideration against the SDK in the first instance?

Reply:
Ahh, ok got it. Thanks!